api/src/test/java/org/openmrs/api/ProgramWorkflowServiceUnitTest.java

```diff
@@ -173,4 +179,146 @@ public void saveProgram_shouldFailIfProgramWorkFlowStateTerminalIsNull() {
 		
 		pws.saveProgram(program);
 	}
+	
+	@Test
+	public void savePatientProgram_shouldFailForNullPatient() {
+		
+		exception.expect(APIException.class);
+		exception.expectMessage("PatientProgram requires a Patient and a Program");
+		
+		PatientProgram patientProgram = new PatientProgram(1);
+		patientProgram.setProgram(new Program(1));
+		
+		pws.savePatientProgram(patientProgram);
+	}
+	
+	@Test
+	public void savePatientProgram_shouldFailForNullProgram() {
+		
+		exception.expect(APIException.class);
+		exception.expectMessage("PatientProgram requires a Patient and a Program");
+		
+		PatientProgram patientProgram = new PatientProgram(1);
+		patientProgram.setPatient(new Patient(1));
+		
+		pws.savePatientProgram(patientProgram);
+	}
+	
+	@Test
+	public void purgePatientProgram_shouldFailForNotEmptyStatesAndTrueCascade() {
+		
+		exception.expect(APIException.class);
+		exception.expectMessage("Cascade purging of PatientPrograms is not implemented yet");
+		
+		PatientProgram patientProgram = new PatientProgram(1);
+		
+		Set<PatientState> patientStates = new Set<PatientState>() {
```

Comment:
this is not needed. you can implement this test as

```java
	@Test
	public void purgePatientProgram_shouldFailGivenNonEmptyStatesAndCascadeSetToTrue() {
		
		exception.expect(APIException.class);
		exception.expectMessage("Cascade purging of PatientPrograms is not implemented yet");	
		
		PatientProgram patientProgram = new PatientProgram();
		PatientState patientState = new PatientState();
		patientProgram.getStates().add(patientState);
		
		pws.purgePatientProgram(patientProgram, true);
	}
```

Reply:
@teleivo This is a much easier way. Thanks